### DISCUSSION TITLE
GitHub Copilot Chat Asked for Too Many Inappropriate Permissions

### DISCUSSION THREAD

**Discussion by leimao** [community/community#107560] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

Recently, I realized that GitHub Copilot Chat starts to ask for more permissions:

![Screenshot from 2024-02-11 08-55-40](https://github.com/community/community/assets/17606112/fada8af0-02dc-4358-81d2-fb909ab00092)

More specifically, it asked me for the READ and **WRITE** access to **ALL** the PUBLIC and **PRIVATE** repositories. I believe this is not acceptable to most of the users.

![Screenshot from 2024-02-11 08-56-24](https://github.com/community/community/assets/17606112/6c169fab-973e-4cc2-835f-1d03184f5eb7)

**Comment by mw-xyz** [comment]
This is for GitHub in general. What that does is allows you to commit, push, pull and edit directly from VSC. Copilot chat itself runs with GitHub. Plus, as it is run by GitHub Inc., it would not do anything dodgy to your repos, as GitHub is of course reputable etc. If you have doubts perhaps read docs on why those perms are needed. Hope this helps

**Comment by Akash1134** [comment]
Hi @leimao ,

You may read about permissions required for GitHub Copilot Chat (VSCode) in [this issue](https://github.com/microsoft/vscode-copilot-release/issues/813). (You might've already contributed, but we want to keep that thread the primary user feedback issue)

- Similar [reports](https://github.com/orgs/community/discussions/106551).
- [GitHub Copilot Pre-release License Terms](https://docs.github.com/en/site-policy/github-terms/github-copilot-pre-release-license-terms)
- [Installing a GitHub App from GitHub Marketplace for your personal account](https://docs.github.com/en/apps/using-github-apps/installing-a-github-app-from-github-marketplace-for-your-personal-account)

Thanks!

**Comment by ppazos** [comment]
It also tries to access all my organizations, this is crazy.

**Comment by twhiteheadzm** [comment]
I am also experiencing this issue. I want to use the chat feature in intellij github copilot plugin but it says I can't use it unless I give it write permissions to all my organization's repositories and there is no clear explanation as to why. Is this a requirement for using the plugin? Do I need to cancel my subscription and find a new provider?

**Comment by Tim-W** [comment]
It would be more appropriate to be able to select specific repositories to give access. This plugin is trying to access organization-level permissions which is holding me back from installing it.
